Source/WebKit/UIProcess/RemoteLayerTree/RemoteScrollingCoordinatorProxy.cpp

```diff
@@ -146,6 +146,11 @@ void RemoteScrollingCoordinatorProxy::currentSnapPointIndicesDidChange(WebCore::
 // This comes from the scrolling tree.
 void RemoteScrollingCoordinatorProxy::scrollingTreeNodeDidScroll(ScrollingNodeID scrolledNodeID, const FloatPoint& newScrollPosition, const std::optional<FloatPoint>& layoutViewportOrigin, ScrollingLayerPositionAction scrollingLayerPositionAction)
 {
+#if ENABLE(INTERACTION_REGIONS_IN_EVENT_REGION)
+    // This is called for all frames (including main).
+    scrollingTreeFixedNodesDidUpdate();
```

Comment:
Why is the scrolledNodeID not passed in, to allow the implementation to be more efficient? Also the name is confusing; we don't know here that a "fixed node did update"; we just know that some scroller scrolled.

Reply:
Is there an efficient way to go from the scrolledNodeId to the list of the impacted "viewport constrained" layers?
(apart from the HashSet global to the coordinator)

We could also hook into `ScrollingStateFixedNode/ScrollingStateStickyNode` directly, but I'm not sure we want to add RemoteLayerTree concerns to `scrolling/`

Reply:
I think hooking up to the scrolling tree is actually the right thing to do here. The scrolling tree knows exactly what layer tree geometry updates are needed when a scroll happens, taking fixed and sticky into account.